mimic/test/test_auth.py

```diff
@@ -774,3 +788,121 @@ def test_impersonation_request_with_invalid_body_causes_http_bad_request(self):
+        """
+        Test to verify :func: `validate_token` and then authenticate
+        """
+        core = MimicCore(Clock(), [ExampleAPI()])
+        root = MimicRoot(core).app.resource()
+
+        (response1, json_body1) = self.successResultOf(json_request(
+            self, root, "GET",
+            "http://mybase/identity/v2.0/tokens/123456a?belongsTo=111111"
+        ))
+        self.assertEqual(200, response1.code)
+        (response, json_body) = self.successResultOf(json_request(
+            self, root, "POST", "/identity/v2.0/tokens",
+            {
+                "auth": {
+                    "tenantId": "111111",
+                    "token": {
+                        "id": "123456a"
+                    }
+                }
+            }
+        ))
+        self.assertEqual(response.code, 200)
+        self.assertEqual(json_body["access"]["token"]["id"],
+                         json_body1["access"]["token"]["id"])
+        self.assertEqual(json_body["access"]["token"]["tenant"]["id"],
+                         json_body1["access"]["token"]["tenant"]["id"])
+        self.assertEqual(json_body["access"]["user"]["name"],
+                         json_body1["access"]["user"]["name"])
+
+    def test_response_for_validate_impersonated_token(self):
+        """
+        Test to verify :func: `validate_token` and then authenticate
+        """
+        core = MimicCore(Clock(), [ExampleAPI()])
+        root = MimicRoot(core).app.resource()
+        # Authenticate the impersonator
+        (response0, json_body0) = self.successResultOf(json_request(
+            self, root, "POST", "/identity/v2.0/tokens",
+            {
+                "auth": {
+                    "tenantId": "111111",
+                    "token": {
+                        "id": "123456a"
+                    }
+                }
+            }
+        ))
+        self.assertEqual(200, response0.code)
+        impersonator_token = json_body0["access"]["token"]["id"]
+
+        # Authenticate using the username so we know the tenant_id
+        (response1, json_body1) = self.successResultOf(json_request(
+            self, root, "POST", "/identity/v2.0/tokens",
+            {
+                "auth": {
+                    "passwordCredentials": {
+                        "username": "test1",
+                        "password": "theUsersPassword"
+                    },
+                    "tenantId": "12345"
+                }
+            }
+        ))
+        self.assertEqual(200, response1.code)
+        # Impersonate user test1
+        (response2, json_body2) = self.successResultOf(json_request(
+            self, root, "POST",
+            "http://mybase/identity/v2.0/RAX-AUTH/impersonation-tokens",
+            {"RAX-AUTH:impersonation": {"expire-in-seconds": 1,
+                                        "user": {"username": "test1"}}},
+            headers={'X-Auth-Token': [str(impersonator_token)]}
```

Comment:
Conceptually, it's expected to be a list because a header might occur multiple times and have multiple values, so using a list allows us to be unambiguous that it only occurs once.